app/services/bill_run.service.js

```diff
@@ -16,23 +16,30 @@ class BillRunService {
   * Note that the updated stats are _not_ saved back to the database; it is up to the caller to do this.
   *
   * @param {Object} transaction translator belonging to the bill run to find and assess
+  * @param {boolean} requestFromGenerateBillRun [false] If true then we are updating the summary within the 'generate bill run'
+  *  process and we should therefore expect the bill run to be in $generating state
   * @returns {module:BillRunModel} a `BillRunModel` if found else it will throw a `Boom` error
   */
-  static async go (transaction) {
+  static async go (transaction, requestFromGenerateBillRun = false) {
     const billRun = await BillRunModel.query().findById(transaction.billRunId)
 
-    this._validateBillRun(billRun, transaction)
+    this._validateBillRun(billRun, transaction, requestFromGenerateBillRun)
     this._updateStats(billRun, transaction)
 
     return billRun
   }
 
-  static _validateBillRun (billRun, transaction) {
+  static _validateBillRun (billRun, transaction, requestFromGenerateBillRun) {
     if (!billRun) {
       throw Boom.badData(`Bill run ${transaction.billRunId} is unknown.`)
     }
 
-    if (!billRun.$editable()) {
+    /**
+     * Test if the bill run is $editable, or if it's being called as part of the bill run generation process, and
+     * reject if it isn't.
+     */
+    const partOfBillRunGeneration = this._partOfBillRunGeneration(billRun, requestFromGenerateBillRun)
+    if (!billRun.$editable() && !partOfBillRunGeneration) {
```

Comment:
I wonder if this logic could be moved into `_partOfBillRunGeneration()` and it be refactored so the call is just something like

```suggestion
    if (!this._updateable(billRun, requestFromGenerateBillRun)) {
```